pkg/mtd/ops_linux.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mtd
+
+import (
+	"os"
+)
+
+// Dev contains information about ongoing MTD status and operation.
+type Dev struct {
+	*os.File
+	devName string
+	data    []byte
+}
+
+// DevName is the default name for the MTD device.
+var DevName = "/dev/mtd"
+
+// NewDev creates a Dev, returning Flasher or error.
+func NewDev(n string) (Flasher, error) {
+	f, err := os.OpenFile(n, os.O_RDWR, 0)
+	if err != nil {
+		return nil, err
+	}
+	return &Dev{File: f, devName: n}, nil
+}
+
+// QueueWrite adds a []byte to the pending write queue.
+func (m *Dev) QueueWrite(b []byte, off int64) (int, error) {
+	return m.File.WriteAt(b, off)
+}
+
+// SyncWrite syncs a pending queue of writes to a device.
+func (m *Dev) SyncWrite() error {
```

Comment:
Is it worth always opening the device with O_SYNC to force writes?

Reply:
In which case QueueWrite should become WriteAt like a regular file

but maybe that's why you did this, so as to not :)

Reply:
No, the issue is that you want to accumulate all the writes, then, once SyncWrite is called, figure out the right combination of blocks to erase, their sizes, and so on. Further, depending on the value of the data, both what's there and what you're writing, no erase may be needed. And that's good because if you can avoid an erase cycle, you can avoid a failed erase cycle ... 

This started out as WriteAt but I realized there's a lot of work to do, given what is desired to be written, and it's hard to know until all the write data is there, and SyncWrite is called. FLASH is just nasty.

Reply:
Can you write that into a comment on the interface?

Reply:
I think this time I got it.